tests/integration/test_integration.py

```diff
@@ -769,42 +769,92 @@ def test_search_revision(self, admin_client, list_name, list_c):
         assert len(results) == 1
         assert results[0].record_list.identifier == list_c.identifier
 
-    def test_search_by_database(self, admin_client, list_name, unresolvable_item, list_c):
+    @pytest.mark.parametrize("include_items", [True, False])
+    def test_search_by_database(
+        self,
+        include_items,
+        admin_client,
+        list_name,
+        two_resolvable_items_in_table,
+        list_c,
+        training_database_guid,
+    ):
         criteria = SearchCriterion(
             name_contains=list_name,
-            contains_records_in_databases=[unresolvable_item.database_guid],
+            contains_records_in_databases=[training_database_guid],
         )
-        results = admin_client.search_for_lists(criteria)
-        assert len(results) == 1
-        assert results[0].record_list.identifier == list_c.identifier
+        results = admin_client.search_for_lists(criteria, include_items=include_items)
+        assert len(results) == 2
+        for result in results:
+            assert result.record_list.identifier == list_c.identifier
+            if include_items:
+                assert len(result.items) == 2
+            else:
+                assert result.items is None
 
-    def test_search_by_multiple_databases(self, admin_client, list_name, unresolvable_item, list_c):
+    @pytest.mark.parametrize("include_items", [True, False])
+    def test_search_by_multiple_databases(
+        self,
+        include_items,
+        admin_client,
+        list_name,
+        two_resolvable_items_in_table,
+        list_c,
+        training_database_guid,
+    ):
         # List of databases = Is in one OR the other
         criteria = SearchCriterion(
             name_contains=list_name,
-            contains_records_in_databases=[unresolvable_item.database_guid, str(uuid.uuid4())],
+            contains_records_in_databases=[training_database_guid, str(uuid.uuid4())],
         )
-        results = admin_client.search_for_lists(criteria)
-        assert len(results) == 1
-        assert results[0].record_list.identifier == list_c.identifier
+        results = admin_client.search_for_lists(criteria, include_items=include_items)
+        assert len(results) == 2
```

Comment:
This is weird, we seemingly get two results back, presumably because the same list matched twice? I'm curious to see if this is resolved by upgrading the server as well.

Reply:
It is. We only expect a single un-duplicated result now.